app/models/push.rb

```diff
@@ -181,23 +181,28 @@ def settings_for_kind
   end
 
   def check_enabled_push_kinds
-    if kind == "file" && !Settings.enable_file_pushes
+    if kind == "file" && !(Settings.enable_logins && Settings.enable_file_pushes)
       errors.add(:kind, I18n.t("pushes.file_pushes_disabled"))
     end
 
-    if kind == "url" && !Settings.enable_url_pushes
+    if kind == "url" && !(Settings.enable_logins && Settings.enable_url_pushes)
       errors.add(:kind, I18n.t("pushes.url_pushes_disabled"))
     end
   end
 
-  def set_note_and_passphrase
+  def set_default_attributes
     self.note ||= ""
     self.passphrase ||= ""
+    self.name ||= ""
   end
 
   def valid_url?(url)
     !Addressable::URI.parse(url).scheme.nil?
   rescue Addressable::URI::InvalidURIError
     false
   end
+
+  def deleted
+    audit_logs.where(kind: AuditLog.kinds[:expire]).exists?
```

Comment:
This is slick

Reply:
:fast_forward:  Credits goes contributors of the Premium one :)